### PR TITLE
Click events

### PR DESCRIPTION
**Pull request by adambraimbridge**
@georgecrawford @markstephens 

Click events: This is a work in progress. It's ready for a first-pass code review and further discussion.

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -1,11 +1,16 @@
 {
   "name": "o-tracking",
-  "main": ["main.js"],
+  "main": [
+    "main.js"
+  ],
```
Comment (adambraimbridge) [inline bower.json]: This was done automatically by bower

File: bower.json [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
     "node_modules",
     "resources",
     "test",
     ".idea"
   ],
-  "description": "Origami module for FT tracking."
+  "description": "Origami module for FT tracking.",
+  "dependencies": {
+    "ftdomdelegate": "~2.0.3"
+  }
```
Comment (adambraimbridge) [inline bower.json]: See https://github.com/ftlabs/ftdomdelegate

**Comment by markstephens** [comment]
How comes the re-write? What changed?

**Comment by adambraimbridge** [comment]
Hi Mark, 
George and I have been re-thinking the way clicks are tracked (specifically for nextFT and web app). There's some documentation about the process here: https://docs.google.com/document/d/18G4oEkfdMbG719YmMnIHeOcD8hVHLk4q9Dr-7pazxRk/ 

I'd like to sit down with you and George at your convenience to discuss this PR and the thinking behind it. We have a 'schemas' meeting scheduled for 15:00 today — that might be a good time, if you can make it? Otherwise we can do a hangout chat with George at some point.

Cheers, Adam

**Comment by georgecrawford** [comment]
You still have a file at `/src/javascript/events/link-click.js`

File: src/javascript/events/click.js [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+const getTrace = el => {
+	const rootEl = document;
+	const clickedEl = el;
+	let trace = [];
+	while (el && el !== rootEl) {
+		let elementProperties = getElementProperties(el);
+
+		// If the element happens to have a data-trackable attribute, get the siblings
+		// and position of the clicked element (relative to the current element).
+		if (elementProperties["data-trackable"]) {
```
Comment (georgecrawford) [inline src/javascript/events/click.js]: Try this, to capture `data-trackable[-anything]`:

```js
Array.from(el.attributes).map(attr => ({name: attr.name, value: attr.value})).filter(attr => attr.name.match(/^data-trackable/))
```

File: src/javascript/events/click.js [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+			const selector = clickedEl.getAttribute('data-trackable') ? `[data-trackable="${clickedEl.getAttribute('data-trackable')}"]` : clickedEl.nodeName;
```
Comment (georgecrawford) [inline src/javascript/events/click.js]: No need to recalculate `selector` each time if it's based off `clickedEl`, which doesn't change.

**Pull request merged**